### ISSUE TITLE
Support for the MSA / MAS / AMAS Hyper-Dimensional Data file?

### ISSUE THREAD

**Issue by ericpre** [hyperspy/hyperspy#1251]
When I was looking around for the specification/documentation of the msa format, I found this pure python package, which implement reading/writing MSA / MAS / AMAS Hyper-Dimensional data files. Although this format looks interesting, I don't know if there are a lot of people using it.
This a pure python package, which is available in pypi, then it should be fairly straightforward to add a 
io plugin in HyperSpy. 

https://github.com/pyhmsa/pyhmsa

http://www.csiro.au/luminescence/HMSA/index.html

**Comment by francisco-dlp** [comment]
I didn't know about this format either, very interesting!

**Comment by jat255** [comment]
This issue should be a proposal and discussion space for the implementation of the MSA HMSA format that was recently announced by Nestor Zaluzec on the `hyperspy-users` mailing list. See the below copy of the email message:

**Sent by Nestor J. Zaluzec <nestor.zaluzec@gmail.com> at 9:36AM:**

> The updated MSA/MAS/AMMS Hyperspectral File format (HMSA)  has been finalized and has been submitted to ISO. This is an update to the original MSA file format which originally only handled single spectra.  This version, designed to promote vendor neutral file sharing, does images, spectra, hyper spectral data, and other multi-dimensional data sets.   It would appropriate to consider  adding  it to HyperSpy. 
>
> The full documentation is available on-line at:
>    https://www.microscopy.org/resources/scientific_data/
>    https://www.microscopy.org/resources/scientific_data/HMSA_Specification-20161100.pdf

>The MSA / MAS / AMAS HyperDimensional Data File (HMSA, for short) is intended to be a common format to permit the exchange of hyper-dimensional microscopy and microanalytical data between different software applications. The expected applications include:
>
>  * Hyper-spectral maps, such as electron energy loss spectroscopy (EELS), energy- dispersive x-ray spectrometry (XEDS), or cathodoluminescence spectroscopy (CL).
>  * ‘Hyper-image’ maps, such as pattern maps using electron backscatter diffraction (EBSD) or convergent beam electron diffraction (CBED).
>  * 3-dimensional maps, such as confocal microscopy, or focused ion beam (FIB) serial section maps.
>  * 4-dimensional maps, such as double-tilt electron tomography.
>  * Time-resolved microscopy and spectroscopy.
>
> In addition to storing hyperdimensional data, the HMSA file format is suitable for storing conventional microscopy and microanalysis data, such as spectra, line profiles, images, and quantitative analyses, as well as experimental conditions and other metadata. 
>
> Nestor Zaluzec
> *Chair of MSA Standards Committee*

**Comment by magnunor** [comment]
Agreed. It would be nice to have these types of files loadable in HyperSpy, and using the `pyhmsa` seems like the easiest solution. However, it doesn't seem to be maintained? Last commit being in May 2016...

**Comment by tjof2** [comment]
Is this the file format listed in the docs already as being supported (https://hyperspy.readthedocs.io/en/latest/user_guide/io.html#emsa-msa), or is it different? If the former, this can be closed @ericpre ?

**Comment by ericpre** [comment]
It is different.

**Comment by tjof2** [comment]
Ok great, thanks!
